### DISCUSSION TITLE
How can I force the public connector URL for broker registration behind a proxy

### DISCUSSION THREAD

**Discussion by trueg** [International-Data-Spaces-Association/DataspaceConnector#120] [Q&A] [answered]
We run a DSC behind a proxy with a public connector URL for communication with other connectors. For setup and administration we use an internal network with the DSC API (imagine something like `http://localhost:8080`).

The problem arises when calling `/api/ids/resource/update` to register resources with the broker from the internal nextwork interface. The registration works properly but the broker will show the internal resource URL `http://localhost:8080/api/...` instead of the public facing one.

Is there a way to force the base URL for the connector during the registration with the broker?

**Comment by HeinrichPet** [comment]
Have you set the right default endpoint? https://international-data-spaces-association.github.io/DataspaceConnector/Deployment/Configuration#step-1-connector-properties 
It is used to provide the external URL to the Metadata Broker.

**Reply by trueg** [reply]
Yes, the default endpoint is set in our config.json:
```
"ids:hasDefaultEndpoint" : {
      "@type" : "ids:ConnectorEndpoint",
      "@id" : "https://ids-dev.corpinter.net/mds/devportal/defaultEndpoint",
      "ids:accessURL" : {
        "@id" : "https://ids-dev.corpinter.net/mds/devportal/api/ids/data"
      }
    }
```
Does the default endpoint id also need to contain a uuid like many other resource URIs do?

**Reply by juliapampus** [reply]
no, it does not need to contain a uuid, the set value looks correct. This has nothing to do with the base url of the resources

**Comment by SebastianOpriel** [comment, marked as answer]
You might be using an NGINX where you have to forward the domain as a header field to the application by adding following line:
`proxy_set_header Host $host;`
